inspire_utils/name.py

```diff
@@ -83,6 +83,9 @@ def __init__(self, name, constants=None):
             self._parsed_name.last = self._parsed_name.first + self._parsed_name.middle + self._parsed_name.suffix
             self._parsed_name.first = ''
 
+    def __iter__(self):
+        return self._parsed_name
```

Comment:
[`__iter__` should return an iterator](https://docs.python.org/2/reference/datamodel.html#object.__iter__), which `self._parsed_name__` isn't. Maybe `iter(self._parsed_name)` works?

Reply:
It actually works because of 
> Iterator objects also need to implement this method; they are required to return themselves. For more information on iterator objects, see Iterator Types.

which actually comes down to 
>Python supports a concept of iteration over containers. This is implemented using two distinct methods; these are used to allow user-defined classes to support iteration. Sequences, described below in more detail, always support the iteration methods.

As `HumanName` provides both `__iter__` and [`__next__`](https://github.com/derek73/python-nameparser/blob/master/nameparser/parser.py#L117) it can support iteration as an "iterator type".

Reply:
:+1: I didn't expect `HumanName` to be already an iterator